repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -73,6 +71,13 @@ private void processCreateProduct(AtlasEntity entity, AtlasVertex vertex) throws
     private void processUpdateProduct(AtlasEntity entity, AtlasVertex vertex) throws AtlasBaseException {
         AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processUpdateProduct");
         String VertexQName = vertex.getProperty(QUALIFIED_NAME, String.class);
+        String productName = (String) entity.getAttribute(NAME);
+        String parentDomainQualifiedName = (String) entity.getAttribute(PARENT_DOMAIN_QN);
```

Comment:
What if while updating, payload does not have `parentDomainQualifiedName` at all
Handle case, in that case we should read it from vertex